Software Saturdays Updates 1/19/23

adithyachan opened the pull request:
Updated everything Hadi mentioned in the old PR and made a new branch

adithyachan committed:
Updated software saturdays landing

adithyachan committed:
Finished up landing

adithyachan committed:
Added Animations and Mentor Page

adithyachan committed:
Smaller photo to comply with <1 MB per page

adithyachan committed:
Changed animation to make it more mobile friendly

adithyachan committed:
Removed unneccesary pages

adithyachan committed:
Fixed link and added box around FAQs

adithyachan committed:
fixed mobile layout

adithyachan committed:
Did everything Hadi mentioned in a new branch

hadiahmed098 commented:
Deployed at https://dev.purdueieee.org/software/

adithyachan commented:
Thanks @hadiahmed098!

hadiahmed098 commented on software/index.html:
```diff
@@ -8,20 +8,21 @@
         <meta name="author" content="Purdue IEEE">
 
-        <title>Purdue IEEE</title>
+        <title>Software Saturdays</title>
```
Don't change this title, the Vue engine already controls it below.

hadiahmed098 commented on software/index.html:
```diff
@@ -36,96 +37,127 @@
+                <!-- Session Alert -->
+                <div class="alert alert-secondary alert-dismissable fade show text-center mt-3" role="alert">
+                    <strong class="fs-3">Registration for Spring 2023 is active. Click <a href="https://forms.gle/C5XQDh6ATyFnwfdX7" class="alert-link" target="_blank" type="link">here</a> to register.</strong>
+                    <button type="button" class="btn-close" data-bs-dismiss="alert" aria-label="Close"></button>
```
This button isn't aligned with the text

hadiahmed098 commented on software/index.html:
```diff
@@ -36,96 +37,127 @@
+                    <img src="./images/logo_white.png" class="pl-3" alt="Software Saturdays Logo" width="350" >
+                    <h1 class="display-1 fw-bold text-light">Software Saturdays</h1>
+                    <div class="container-fluid">
+                        <img src="./images/IEEE_white.png" alt="IEEE Logo" width="200">
```
These images are a bit small, also you are mixing the wrong types of logos: the PIEEE logo is a horizontal text while the CoE logo is a vertical text. Unify these to be the same type.

hadiahmed098 commented on software/index.html:
```diff
@@ -36,96 +37,127 @@
-                        <img src="./images/logo_white.jpg" alt="Software Saturdays Logo" width="300">
-                        <h1 class="display-3 fw-bold">Software Saturdays</h1>
-                        <p class="fs-4">Sponsored by the Purdue College of Engineering</p>
+                     <!-- Registration Buttons -->
```
The buttons could still be more prominent, this is the only interaction on the page.

hadiahmed098 commented on software/index.html:
```diff
@@ -36,96 +37,127 @@
-                    <p class="fs-5 text-start">
-                        Sponsored by the Purdue College of Engineering, we are able to offer an official Purdue College of Engineering certification for everyone who attends!
+             <!-- Basic Info -->
+            <div class="my-3 p-5 container-fluid fade-in-left">
```
Wide text does not look good on a large screen, it's hard to scan back and forth. Try a responsive container.

hadiahmed098 commented on software/index.html:
```diff
@@ -36,96 +37,127 @@
+                <div class="d-flex flex-row justify-content-between">
+                    <img src="images/react_logo.svg" width="30%" alt="ReactJS Logo" >
+                    <div class="d-flex flex-column">
+                        <h2 class="fs-1 text-light text-end">Spring 2023: <strong class="react-text">React</strong>.</h2>
```
Right-aligned text is hard to read, keep text naturally aligned to the left. Look at the home page for an example.

hadiahmed098 commented on software/index.html:
```diff
@@ -36,96 +37,127 @@
+             <!-- Semester Info -->
+            <div class="my-2 p-5 container-fluid text-light fade-in-right">
+                <div class="d-flex flex-row justify-content-between">
+                    <img src="images/react_logo.svg" width="30%" alt="ReactJS Logo" >
```
This image does not scale at all on mobile

hadiahmed098 commented on software/index.html:
```diff
@@ -36,96 +37,127 @@
+             <!-- Basic Info -->
+            <div class="my-3 p-5 container-fluid fade-in-left">
+                <div class="d-flex flex-column">
+                    <h2 class="fs-1 text-light">Software Development <strong class="text-secondary">Simplified</strong>.</h2>
```
The word "Simplified" has bad contrast to the background.

hadiahmed098 commented on software/index.html:
```diff
@@ -36,96 +37,127 @@
+                <div class="container-fluid bg-secondary mentor-page mb-5">
+                    <div class="d-flex flex-column pt-5">
+                        <div class="d-flex flex-row justify-content-center">
+                            <img src="/images/officers/2022_Adithya_Chandrasekar.jpg" class="circle-img-frame mb-4" width="60%" alt="ReactJS Logo">
```
Wrong alt-text, make the image tag smaller so it looks better.

hadiahmed098 commented on software/index.html:
```diff
@@ -36,96 +37,127 @@
+                    <strong>Q: How much prior experience do I need?</strong> <br/> 
+                    <small class="text-muted">A: Absolutely none. We will teach you what you have to know. If you are familiar with web development, we do offer an intermediate track that may be more your style.</small> <br/>
+                    <strong>Q: What is this certificate you mentioned?</strong> <br/> 
+                    <small class="text-muted">A: Any attendee who completes our final project will recieve a digital certificate from the College of Engineering, with a QR code to verify your achievement.</small>
```
Watch out for contrast on the "text-muted" color and the Purdue background.

hadiahmed098 commented on software/index.html:
```diff
@@ -36,96 +37,127 @@
+                        <h2 class="fs-1">Adithya Chandrasekar</h2>
+                        <h5><strong>Software Saturdays Chair & Head Mentor</strong></h3>
+                    </div>
+                    <div class="d-block justify-content-center pb-5">
```
The "Contact me:" portion looks oddly small compared to the rest of the page.

hadiahmed098 replied:
Still unresolved.

adithyachan replied:
Added an fs-4

hadiahmed098 reviewed (changes requested):
Went through it again with a finer tooth comb. Once again, I won't comment on the look of the page compared to the rest of the site.

I would recommend adding more text, it looks pretty bare and not nearly descriptive enough about what the program is.

hadiahmed098 commented:
@Rmalavathu Please also review this and get everything resolved quickly, Adithya will need this to be ready for marketing ASAP.

Rmalavathu commented on software/index.html:
```diff
@@ -36,96 +37,127 @@
+                    <h2 class="fs-1 text-light">Software Development <strong class="text-secondary">Simplified</strong>.</h2>
+                    <br/>
+                    <p class="fs-4 text-start text-light">
+                         Software Satrudays is a free 8-week course, sponsered by Purdue IEEE and the College of Engineering, covering the basics of a software development framework, rotating every semester. This program is for any Purdue student who is seeking to learn a new software development framework, spruce up their resume, or is just interested in software. Additionally, everyone can earn ...
```
Fix up the spelling like saturdays and sponsored

Rmalavathu commented on software/index.html:
```diff
@@ -36,96 +37,127 @@
+
+             <!-- FAQs -->
+            <div class="my-5 p-5 fs-3 container-fluid d-block text-start text-light bg-purdue extra-rounded">
+                    <h2><strong>FAQs:</strong></h3>
```
For me, it is a little weird how FAQ doesn't really fade in compared to the other elements which seems off compared to the other behaviors of the website

Rmalavathu reviewed (commented):
Overall it is good, but there are some small issues which hadi pointed out. The big ones to me were the alignment of the buttons, and the small contact section compared to the rest of the page. One thing I would suggest to add is a way to see the past project like the old version of the page had.

adithyachan committed:
Fixed more of the things Hadi mentioned

adithyachan commented:
@Rmalavathu I agree with you on everything and I'll get it fixed. Regarding previous projects, I am going to update that later as I really need to get the updated home page out so that people can access the forms.

adithyachan committed:
Updated FAQs animation and contact size

adithyachan commented:
@Rmalavathu @hadiahmed098 These are the changes I made in the last two commits:

- I have removed the react logo because I need some time to figure out how to get it to scale properly on mobile. 
- I added a sponsors section because the IEEE logo and the COE logo were kinda oddly placed up top. 
- I added max-widths on the para text so it doesn't span the whole page on desktop
- I made the buttons and contact info significantly larger
- Added animations to all the sections

Lmk if there is anything else I should edit as I would like to get this live ASAP

Rmalavathu commented:
Deployed at https://dev.purdueieee.org/software/

Rmalavathu commented:
These changes look good and don't have any major issues besides the past projects which we can add later

hadiahmed098 commented on software/index.html:
```diff
@@ -36,96 +37,139 @@
+                    <h2 class="fs-1 text-light">Software Development <strong class="purdue-text">Simplified</strong>.</h2>
+                    <br/>
+                    <p class="fs-4 text-start text-light para-width">
+                         Software Saturdays is a free 8-week course, sponsored by Purdue IEEE and the College of Engineering, covering the basics of a software development framework, rotating every semester. This program is for any Purdue student who is seeking to learn a new software development framework, spruce up their resume, or is just interested in software. Additionally, everyone can earn ...
```
This is begging for an image on the right, it looks off without it.

adithyachan replied:
Added image and had it get removed for mobile since it just doesn't fit well on a mobile page

hadiahmed098 commented on software/index.html:
```diff
@@ -36,96 +37,139 @@
+            <div class="my-2 p-5 container-fluid text-light fade-in-right">
+                <div class="d-flex flex-row justify-content-end">
+                    <div>
+                        <h2 class="fs-1 text-light text-end">Spring 2023: <strong class="react-text">React</strong>.</h2>
```
This header should be aligned with the start of the text below it.

adithyachan replied:
Changed the text alignment

hadiahmed098 commented on software/index.html:
```diff
@@ -36,96 +37,139 @@
+                <div class="d-flex flex-row justify-content-end">
+                    <div>
+                        <h2 class="fs-1 text-light text-end">Spring 2023: <strong class="react-text">React</strong>.</h2>
+                        <br/>
+                        <p class="fs-4 text-start para-width">
+                            Join us to learn ReactJS in a series of 7 in-depth lessons. Covering the fundamentals of HTML, CSS, JS, and of course, <strong class="react-text">React</strong>. Put your skills to the test while building a full-stack web application using the <strong class="poke-text">Poke</strong>API and <strong class="firebase-text">Firebase</strong>. We currently have a beginner an ...
```
This section is begging for an image on the left, it looks blank without it.

adithyachan replied:
Added image and had it get removed for mobile since it just doesn't fit well on a mobile page

hadiahmed098 commented on software/index.html:
```diff
@@ -36,96 +37,139 @@
+                        <img class="w-50" src="./images/coe_logo.png" alt="College of Engineering Logo">
+                    </div>
+                    <div class="col">
+                        <img class="w-50" src="./images/IEEE_logo.png" alt="IEEE Logo">
```
PIEEE is not a sponsor, they are the hosting organization.

Also, minor nitpick but the colors are not matched - I'll get Alex to send you a variation where the icon is colored and the text is black.

adithyachan replied:
Hmm I am think I will remove the PIEEE logo since it is technically on the PIEEE website so that would just be redundant

adithyachan replied:
Moved engineering logo next to first para, I think this is a better place and removes the whole sponsor section

hadiahmed098 commented on software/index.html:
```diff
@@ -36,96 +37,139 @@
+            </div>
+
+             <!-- FAQs -->
+            <div class="my-5 p-5 fs-3 container-fluid d-block text-start text-light bg-purdue extra-rounded fade-in-left">
```
White on gold is not a good contrast.

adithyachan replied:
Changed to black

hadiahmed098 commented on software/index.html:
```diff
@@ -36,96 +37,139 @@
+            <div class="py-5 my-2 container-lg bg-dark">
+                <!-- Session Alert -->
+                <div class="alert alert-secondary alert-dismissable fade show text-center mt-3" role="alert">
+                    <strong class="fs-3">Registration for Spring 2023 is active. Click <a href="https://forms.gle/C5XQDh6ATyFnwfdX7" class="alert-link" target="_blank" type="link">here</a> to register.</strong>
```
Web Accessibility guidelines state that a link needs to be more descriptive than just "here". It should explain what the user is clicking on.

adithyachan replied:
Added an aria label

hadiahmed098 commented on software/index.html:
```diff
@@ -36,96 +37,139 @@
+                <!-- Session Alert -->
+                <div class="alert alert-secondary alert-dismissable fade show text-center mt-3" role="alert">
+                    <strong class="fs-3">Registration for Spring 2023 is active. Click <a href="https://forms.gle/C5XQDh6ATyFnwfdX7" class="alert-link" target="_blank" type="link">here</a> to register.</strong>
+                    <button type="button" class="btn-close" data-bs-dismiss="alert"></button>
```
This button is still misaligned.

adithyachan replied:
I am confused about what is misaligned. It is aligned with the text when I see it?

adithyachan replied:
center aligned the button and justified it to the end

hadiahmed098 commented on software/index.html:
```diff
@@ -36,96 +37,139 @@
+                        <img class="w-50" src="./images/coe_logo.png" alt="College of Engineering Logo">
+                    </div>
+                    <div class="col">
+                        <img class="w-50" src="./images/IEEE_logo.png" alt="IEEE Logo">
```
On mobile, these images do not scale properly. Try to have them stack vertically with a flex box once they are below a threshold. Look at https://dev.purdueieee.org/rov/sponsors/.

adithyachan committed:
more fixes, added images, removed sponsors

hadiahmed098 committed:
Fixed image wrapping and reorganized some divs

hadiahmed098 committed:
Logo resizes properly

hadiahmed098 commented:
This should now be ready to go.

adithyachan commented:
Thanks for the help @hadiahmed098
